staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1091 @@
+		for i, err := 1, ca.netFabric.DeleteRemoteIfc(aPreExistingRemoteIfc); err != nil; i++ {
+			glog.V(3).Infof("Init: deletion of orphan remote interface %#+v failed: %s. Attempt nbr. %d",
+				aPreExistingRemoteIfc,
+				err.Error(),
+				i)
+			time.Sleep(netFabricRetryPeriod)
+		}
+		glog.V(3).Infof("Init: deleted orphan local interface %#+v", aPreExistingRemoteIfc)
+	}
+
+	return nil
+}
+
+func (ca *ConnectionAgent) processQueue() {
+	for {
+		item, stop := ca.queue.Get()
+		if stop {
+			return
+		}
+		attNSN := item.(k8stypes.NamespacedName)
+		ca.processQueueItem(attNSN)
+	}
+}
+
+func (ca *ConnectionAgent) processQueueItem(attNSN k8stypes.NamespacedName) {
+	defer ca.queue.Done(attNSN)
+	err := ca.processNetworkAttachment(attNSN)
+	requeues := ca.queue.NumRequeues(attNSN)
+	if err != nil {
+		// If we're here there's been an error: either the attachment current state was
+		// ambiguous (e.g. more than one vni), or there's been a problem while processing
+		// it (e.g. Interface creation failed). We requeue the attachment reference so that
+		// it can be processed again and hopefully next time there will be no errors.
+		glog.Warningf("Failed processing NetworkAttachment %s, requeuing (%d earlier requeues): %s",
+			attNSN,
+			requeues,
+			err.Error())
+		ca.queue.AddRateLimited(attNSN)
+		return
+	}
+	glog.V(4).Infof("Finished NetworkAttachment %s with %d requeues", attNSN, requeues)
+	ca.queue.Forget(attNSN)
+}
+
+func (ca *ConnectionAgent) processNetworkAttachment(attNSN k8stypes.NamespacedName) error {
+	att, deleted, err := ca.getAttachment(attNSN)
+	if att != nil {
+		// If we are here the attachment exists and it's current state is univocal
+		err = ca.processExistingAtt(att)
+	} else if deleted {
+		// If we are here the attachment has been deleted
+		err = ca.processDeletedAtt(attNSN)
+	}
+	return err
+}
+
+// getAttachment attempts to determine the univocal version of the NetworkAttachment with namespaced name attNSN. If it succeeds it
+// returns the attachment if it is found in an Informer cache or attDeleted set to true if it could not be found in any cache (e.g.
+// because it has been deleted). If the current attachment version cannot be determined without ambiguity, an error is returned.
+// An attachment is considered amibguous if it either has been seen with more than one vni in a remote attachments cache, or if
+// it is found both in the local attachments cache and a remote attachments cache.
+func (ca *ConnectionAgent) getAttachment(attNSN k8stypes.NamespacedName) (att *netv1a1.NetworkAttachment, attDeleted bool, err error) {
+	// Retrieve the VNIs where the attachment could be as a remote attachment
+	attSeenInVNIs := ca.getAttVNIs(attNSN)
+	nbrOfVNIs := len(attSeenInVNIs)
+	if nbrOfVNIs > 1 {
+		// If the attachment could be a remote one in more than one VNI, return an error. This will cause
+		// the att reference to be requeued, and hopefully by the time it is processed again the
```

Comment:
We do not even need to requeue in this case; the deletion that brings the set size down to 1 will enqueue.

Reply:
Fixed in 5aca980.